nslsii/temperature_controllers.py

```diff
@@ -0,0 +1,85 @@
+from ophyd import DeviceStatus
+from ophyd.mixins import EpicsSignalPositioner
+from time import sleep
+
+
+class Eurotherm(EpicsSignalPositioner):
+    '''This class is used for integrating with Eurotherm controllers.
+
+    Parameters
+    ----------
+    PV_prefix : str.
+        The PV prefix that is common to the readback and setpoint PV's.
+    tolerance : float, optional.
+        The range of temperature within which it can be considered constant.
+        Default is 1.
+    equilibrium_time : float, optional.
+        The time (in seconds) that a temperature should be within `tolerance`
+        for it to be considered at equilibrium. Default is 5 seconds.
+    timeout : float, optional.
+        The maximum time (in seconds) to wait for the temperature to reach
+        equilibrium before raising a TimeOutError. Default is 30 seconds.
+    PV_read_suffix, PV_write_suffix : str, optional
+        The suffix for the readback and setpoint PV's respectively. Defaults
+        are `'-I'` and `'-SP'` respectively.
+    '''
+
+    def __init__(self, PV_prefix, tolerance=1, equilibrium_time=5,
+                 timeout=30, PV_read_suffix='-I', PV_write_suffix='-SP',
+                 **kwargs):
+        self.read_pv = PV_prefix+PV_read_suffix
+        self.write_pv = PV_prefix+PV_write_suffix
+        super().__init__(read_pv=self.read_pv, write_pv=self.write_pv,
+                         tolerance=tolerance, **kwargs)
+        self._equilibrium_time = equilibrium_time
+        self._timeout = timeout
+
+    # Setup some new signals required for the moving indicator logic
+    # This approach mirrors that used for tolerance in ophyd/signal/Signal
+    @property
+    def equilibrium_time(self):
+        '''The time that a temperature should be within `tolerance` for it to
+        be considered at equilibrium.
+        '''
+        return self._equilibrium_time
+
+    @equilibrium_time.setter
+    def equilibrium_time(self, equilibrium_time):
```

Comment:
We have 3 different objects called `equilibrium_time` - a kwarg in the `__init__(...)`, a function, and a kwarg for this function. Will we see any name collisions?

Reply:
That is an excellent question @mrakitin. this structure (and the 3 different objects with the same name) mirrors what is found for `tolerance` in [ophyd/signal/Signal](https://github.com/NSLS-II/ophyd/blob/112c88968e1965803868195818d4ae122b742ee0/ophyd/signal.py#L91) and does not create collisions there. 

I like using the same name as it makes things a little easier to follow (IMO) but am also open to changing it it is considered to risky.

Reply:
OK, makes sense to me, thanks for the reference, @awalter-bnl!

Reply:
We already have a 'settle_time' on positioners, we probably should re-use that name here.

Reply:
This is different from a settle time. A settle time is how long to wait after motion is complete to wait before moving on. This is how long to wait before rechecking that the temperature is within the tolerance range, and hence the temperature is 'stable'. I htink it makes sense to use a different term here as they are different values.